contracts/SlashIndicator.sol

```diff
@@ -190,21 +190,19 @@ contract SlashIndicator is ISlashIndicator,System,IParamSubscriber, IApplication
     if (finalitySlashRewardRatio == 0) {
       finalitySlashRewardRatio = INIT_FINALITY_SLASH_REWARD_RATIO;
     }
-    uint256 srcNumA = _evidence.voteA.srcNum;
-    uint256 tarNumA = _evidence.voteA.tarNum;
-    uint256 srcNumB = _evidence.voteB.srcNum;
-    uint256 tarNumB = _evidence.voteB.tarNum;
 
     // Basic check
-    require(srcNumA+256 > block.number && srcNumB+256 > block.number, "too old block involved in the evidence");
+    require(_evidence.voteA.srcNum+256 > block.number &&
+      _evidence.voteB.srcNum+256 > block.number, "too old block involved in the evidence");
     require(!(_evidence.voteA.srcHash == _evidence.voteB.srcHash &&
     _evidence.voteA.tarHash == _evidence.voteB.tarHash), "two identical votes");
-    require(srcNumA < tarNumA && srcNumB < tarNumB, "source number bigger than target number");
+    require(_evidence.voteA.srcNum < _evidence.voteA.tarNum &&
+      _evidence.voteB.srcNum < _evidence.voteB.tarNum, "source number bigger than target number");
 
     // Vote rules check
-    if (!((srcNumA<srcNumB && tarNumB<tarNumA) || (srcNumB<srcNumA && tarNumA<tarNumB)) && tarNumA != tarNumB) {
-      revert(string(abi.encodePacked("no violation of vote rules")));
-    }
+    require((_evidence.voteA.srcNum<_evidence.voteB.srcNum && _evidence.voteB.tarNum<_evidence.voteA.tarNum) ||
+    (_evidence.voteB.srcNum<_evidence.voteA.srcNum && _evidence.voteA.tarNum<_evidence.voteB.tarNum) ||
```

Comment:
The second line needs to be indented， and the same to the third line.

Reply:
Fixed